numpy/core/_internal.py

```diff
@@ -161,9 +161,10 @@ def _commastring(astr):
         mo = format_re.match(astr, pos=startindex)
         try:
             (order1, repeats, order2, dtype) = mo.groups()
-        except (TypeError, AttributeError):
-            raise ValueError('format number %d of "%s" is not recognized' %
-                                            (len(result)+1, astr))
+        except (TypeError, AttributeError) as e:
+            raise ValueError(
+                f'format number {len(result)+1} of "{astr}" is not recognized'
+                ) from e
```

Comment:
Also should be `from None`, the error here isn't interesting

Reply:
@eric-wieser  I will make the necessary corrections.